meteoblue_dataset_sdk/caching/filecache.py

```diff
@@ -0,0 +1,105 @@
+import datetime
+import hashlib
+import json
+import logging
+import os
+import tempfile
+import zlib
+from typing import Union, Optional
+
+import aiofiles
+from aiofiles import os as aios
+
+from .cache import Cache
+
+CACHE_DIR = "mb_cache"
+# 7200s == 2h
+DEFAULT_CACHE_DURATION = 7200
+
+
+class FileCache(Cache):
+    def __init__(
+        self,
+        cache_path: Optional[str] = None,
+        cache_ttl: int = DEFAULT_CACHE_DURATION,
+        compression_level: int = 6,
+    ):
+        if cache_path is None:
+            cache_path = tempfile.gettempdir()
+        cache_path = os.path.join(cache_path, CACHE_DIR)
+        if not os.path.exists(cache_path):
+            os.makedirs(cache_path)
+
+        self.cache_path = cache_path
+        self.cache_ttl = cache_ttl
+        self.compression_level = compression_level
+
+    async def set(self, query_params: dict, value: bytes) -> None:
+        """ "
+        Hash the query params and use its value a directory and file name.
+        If there is already a valid cache file existing, it will exit. Otherwise it
+        will write to a temporary file before renaming it.
+        :param query_params: query parameters of the query to be stored.
+        :param value: The request data to store
+        """
+        if not query_params:
+            return
+        dir_name, file_name = self._params_to_path_names(query_params)
+        dir_path = os.path.join(self.cache_path, dir_name)
+        file_path = os.path.join(dir_path, file_name)
+        if not os.path.exists(dir_path):
+            await aios.mkdir(dir_path)
+        if self._is_cached_file_valid(file_path):
```

Comment:
There is an error in this logic: If a file exists and is expired, it will never be cached, because there is a return here. `set` should not care if the file exists or is expired.